packages/core/src/typeschema/validation.ts

```diff
@@ -113,46 +113,44 @@ class ResourceValidator {
   }
 
   private checkProperty(parent: TypedValue, key: string, schema: InternalTypeSchema, path: string): void {
-    const propertyValues = getNestedProperty(parent, key);
+    const value = getTypedPropertyValue(parent, key);
```

Comment:
By replacing `getNestedProperty` with a direct call to `getTypedPropertyValue`, you can avoid the outer array, and avoid the for loop on old line 121.